rex/multi_res_resource.py

```diff
@@ -0,0 +1,340 @@
+        ds, ds_slice = parse_keys(keys)
+        _, ds_name = os.path.split(ds)
+        base_name, _ = self._parse_name(ds_name)
+
+        hr_heights = getattr(self._hr_res, '_interpolation_variable', {})
+        lr_heights = getattr(self._lr_res, '_interpolation_variable', {})
+        hr_heights = hr_heights.get(base_name, [])
+        lr_heights = lr_heights.get(base_name, [])
+
+        if ds_name.startswith('time_index'):
+            out = self._hr_res._get_time_index(ds, ds_slice)
+
+        elif ds_name.startswith('meta'):
+            out = self._hr_res._get_meta(ds, ds_slice)
+
+        elif ds_name.startswith('coordinates'):
+            out = self._hr_res._get_coords(ds, ds_slice)
+
+        elif 'SAM' in ds_name:
+            msg = ('SAM dataframe retrieval not implemented for '
+                   'MultiResolutionResource, use '
+                   'MultiResolutionResource.preload_SAM() instead.')
+            logger.error(msg)
+            raise NotImplementedError(msg)
+
+        elif ds_name in self._hr_res.dsets or any(hr_heights):
+            out = self._hr_res._get_ds(ds, ds_slice)
+
+        elif ds_name in self._lr_res.dsets or any(lr_heights):
+            ds_slice = self.map_ds_slice(ds_slice)
+            out = self._lr_res._get_ds(ds, ds_slice)
+            out = self.time_interp(out)
+
+        return out
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self._i >= len(self.datasets):
+            self._i = 0
+            raise StopIteration
+
+        dset = self.datasets[self._i]
+        self._i += 1
+
+        return dset
+
+    def __contains__(self, dset):
+        return dset in self.datasets
+
+    def __getattr__(self, attr):
+        if attr in dir(self):
+            return getattr(self, attr)
+        if attr in self.HR_ATTRS:
+            return getattr(self._hr_res, attr)
+        else:
+            try:
+                hr_attr = getattr(self._hr_res, attr)
+                lr_attr = getattr(self._lr_res, attr)
+                if hasattr(hr_attr, '__call__'):
+                    return hr_attr
+                elif isinstance(hr_attr, list) and isinstance(lr_attr, list):
+                    return list(set(hr_attr + lr_attr))
+                elif isinstance(hr_attr, tuple) and isinstance(lr_attr, tuple):
+                    return hr_attr + lr_attr
+                elif isinstance(hr_attr, dict) and isinstance(lr_attr, dict):
+                    out = copy.deepcopy(lr_attr)
+                    out.update(hr_attr)
+                    return out
+            except Exception as e:
+                msg = ('Could not retrieve attribute "{}" from '
+                       'MultiResolutionResource handler, the hr and lr '
+                       'handler attributes could not be combined: {} {}'
+                       .format(attr, hr_attr, lr_attr))
+                logger.error(msg)
+                raise RuntimeError(msg) from e
```

Comment:
Any particular reason for `RuntimeError` here? I think generally `__getattr__` raises `AttributeError`

Reply:
Agree, forgot about attrerror, I usually just default to runtimeerror haha